includes/EmailNotification.php

```diff
@@ -266,43 +314,90 @@ private function actuallyNotifyOnPageChange( $watchers ) {
 
 		$this->setReplyToAndFromAddresses();
 
-		# The following code is only run, if several conditions are met:
-		# 1. EmailNotification for pages (other than user_talk pages) must be enabled
-		# 2. minor edits (changes) are only regarded if the global flag indicates so
-		if ( !$this->isMinorEdit() || ( $this->notifyUsersOnMinorEdits() && $this->editorWantsToNotifyOnMinorEdits() ) ) {
```

Comment:
Moved this logic into the method `editIsSignificant` and then negated the if statement so we can exit at the top of the function and not wrap almost all of this function in an IF block.